ttnn/cpp/ttnn/tensor/xtensor/conversion_utils.cpp

```diff
@@ -0,0 +1,163 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "common/assert.hpp"
+#include "ttnn/cpp/ttnn/operations/copy.hpp"
+#include "ttnn/tensor/host_buffer/functions.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/xtensor/conversion_utils.hpp"
+#include <ttnn/tensor/xtensor/xtensor_all_includes.hpp>
+
+namespace ttnn::experimental::xtensor {
+namespace {
+
+using ::tt::tt_metal::Tensor;
+
+// copypaste from deprecated tensor pybinds ttnn
+tt::tt_metal::OwnedBuffer create_owned_buffer(const std::vector<float>& data, DataType data_type) {
+    using ::tt::tt_metal::owned_buffer::create;
+
+    switch (data_type) {
+        case DataType::BFLOAT8_B: {
+            auto uint32_vector = pack_fp32_vec_as_bfp8_tiles(data, /*row_major_input=*/false, /*is_exp_a=*/false);
+            return create<uint32_t>(std::move(uint32_vector));
+        }
+        case DataType::BFLOAT4_B: {
+            auto uint32_vector = pack_fp32_vec_as_bfp4_tiles(data, /*row_major_input=*/false, /*is_exp_a=*/false);
+            return create<uint32_t>(std::move(uint32_vector));
+        }
+        case DataType::FLOAT32: {
+            auto data_copy = data;
+            return create<float>(std::move(data_copy));
+        }
+        case DataType::BFLOAT16: {
+            std::vector<bfloat16> bfloat16_data(data.size());
+            std::transform(std::begin(data), std::end(data), std::begin(bfloat16_data), [](float value) {
+                return bfloat16(value);
+            });
+            return create<bfloat16>(std::move(bfloat16_data));
+        }
+        default: {
+            TT_THROW("Cannot create a host buffer for data type: {}", data_type);
+        }
+    }
+}
+
+template <typename T>
+Tensor create_owned_tensor(
+    std::vector<T> data, const ttnn::Shape& shape, tt::tt_metal::DataType data_type, tt::tt_metal::Layout layout) {
```

Comment:
no ttnn::Shape please. anything prevents us from making this ttnn::SimpleShape?

Reply:
must be `Tensor(vector raw_data,  const TensorSpec&)` 
raw data in this case would mean that some transformation must happen to align raw_data to the requested spec.

we can additionally have `Tensor(Data data,  const TensorSpec&)` for cases where we want to create a Tensor from a data which was already processed and user knows its aligned with the spec. Data in this case is just a type that provides a semantic distinction between raw and prepared vector.

Reply:
> anything prevents us from making this ttnn::SimpleShape?

No reason - had this change locally:)

+1 to `TensorSpec`. I'm adding some documentation to clarify that the input has to match the tensor volume, and that we are assuming the row major layout.

For the second use case where we have the prepared / pre-processed data, we can do this in a follow up, this seems a somewhat separate effort.